src/protocol/boolean/dumb_bitwise_lt.rs

```diff
@@ -0,0 +1,155 @@
+    ) -> Result<Vec<Replicated<F>>, Error> {
+        let less_than = zip(a, b).enumerate().rev().map(|(i, (a_bit, b_bit))| {
+            let c = ctx.narrow(&BitOpStep::Step(i));
+            let one = c.share_of_one();
+            async move { c.multiply(record_id, &(one - a_bit), b_bit).await }
+        });
+        try_join_all(less_than).await
+    }
+
+    ///
+    /// Compares the `a` and `b`, and returns `1` iff `a < b`
+    /// This logic is very simple:
+    /// Starting from the most-significant bit and working downwards:
+    /// For the most-significant bit: check if `a_0 == 0 && b_0 == 1`
+    /// For any the other bit `j`: check if `a_i == b_i` for `i = 0..j-1` AND `(a_j == 0 && b_j == 1)`
+    /// Finally, since at most one of these conditions can be true, it is sufficient to just add up
+    /// all of these conditions. That sum is logically equivalent to an OR of all conditions.
+    #[allow(dead_code)]
+    #[allow(clippy::many_single_char_names)]
+    pub async fn execute<F: Field>(
+        ctx: SemiHonestContext<'_, F>,
+        record_id: RecordId,
+        a: &[Replicated<F>],
+        b: &[Replicated<F>],
+    ) -> Result<Replicated<F>, Error> {
+        debug_assert_eq!(a.len(), b.len(), "Length of the input bits must be equal");
+        let (xored_bits, less_thaned_bits) = try_join(
+            Self::xor_all_but_the_last_bit(a, b, ctx.narrow(&Step::BitwiseAXorB), record_id),
+            Self::less_than_all_bits(a, b, ctx.narrow(&Step::BitwiseALessThanB), record_id),
+        )
+        .await?;
+
+        let one = ctx.share_of_one();
+        let mut any_condition_met = less_thaned_bits[0].clone();
+        let mut all_preceeding_bits_the_same = one.clone() - &(xored_bits[0]);
+        let check_each_bit_context = ctx.narrow(&Step::CheckEachBit);
+        let prefix_equal_context = ctx.narrow(&Step::PrefixEqual);
+        for i in 1..(less_thaned_bits.len() - 1) {
+            let (ith_bit_condition, prefix) = try_join(
+                check_each_bit_context.narrow(&BitOpStep::Step(i)).multiply(
+                    record_id,
+                    &less_thaned_bits[i],
+                    &all_preceeding_bits_the_same,
+                ),
+                prefix_equal_context.narrow(&BitOpStep::Step(i)).multiply(
+                    record_id,
+                    &(one.clone() - &xored_bits[i]),
+                    &all_preceeding_bits_the_same,
+                ),
+            )
+            .await?;
+            all_preceeding_bits_the_same = prefix;
+            any_condition_met += &ith_bit_condition;
+        }
+        let final_index = a.len() - 1;
+        let final_bit_condition = check_each_bit_context
+            .narrow(&BitOpStep::Step(final_index))
+            .multiply(
+                record_id,
+                &less_thaned_bits[final_index],
+                &all_preceeding_bits_the_same,
+            )
+            .await?;
+        any_condition_met += &final_bit_condition;
```

Comment:
The ith_bit _condition will have a value of up to n-1, which could be a problem if we are assuming that it is 0 or 1.

Reply:
Ah, but it will always be either 0 or 1. This is a very clever little trick that's inspired by the original paper.

Logically, it's not possible for more than one of these conditions to be true. They can all be false (zero), or at most one can be true (one). 

The conditions are:
- The most significant bit is less
- the first bits are equal and the second bit is less
- the first two bits are equal the the 3rd bit is less
- the first three bits are equal and the 4th bit is less
- ...